pkg/controllers/termination/eviction.go

```diff
@@ -0,0 +1,87 @@
+package termination
+
+import (
+	"context"
+	"sync"
+
+	set "github.com/deckarep/golang-set"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/api/policy/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/util/workqueue"
+)
+
+type EvictionQueue struct {
+	queue        workqueue.RateLimitingInterface
+	coreV1Client corev1.CoreV1Interface
+
+	enqueued set.Set
+	once     sync.Once
+}
+
+// Evict adds a pod to the EvictionQueue
+func (e *EvictionQueue) Add(pods []*v1.Pod) {
+	// Start processing eviction queue if it hasn't started already
+	e.once.Do(func() { go e.run() })
+
+	for _, pod := range pods {
+		nn := types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}
+		if !e.enqueued.Contains(nn) {
```

Comment:
define this inline
```
if nn := ...; e.enqueued.Contains(nn) {
...
}
```

Reply:
Thanks. Fun fact, since `types.NamespacedName{}` is init a new struct, we actually need to surround it by parentheses or the compiler will think it's the end of the if statement condition!

```
if nn := (types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}); !e.enqueued.Contains(nn) {
...
}
```